torch/_inductor/codegen/rocm/rocm_kernel.py

```diff
@@ -211,6 +214,21 @@ def size(
         val = sympy_product(sizes)
         return cexpr(self.rename_indexing(val))
 
+    def contiguous_stride(self, node: IRNode, default_value: int = 0) -> str:
+        """
+        Hook called from template call to get the contiguous stride of an arg.
+        """
+        # strides: in torch, stride measures how many positions to advance the data pointer when the dimension index is incremented
+        # in CK, the stride is the size of the contiguous dimension
+        # so depending whether input is regular or transposed,
+        # the CK stride is either in -2 or -1 position of the strides list
+        # the below should do the job for 2D inputs
+        if node is None:
+            return str(default_value)
+
+        contiguous_stride = functools.reduce(lambda a, b: max(b, a), node.get_stride(), sympy.Integer(0))
```

Comment:
Does Python `max` work well with symbolic shapes? Or should we use something internal (from `experimental/symbolic_shapes.py`)?

Reply:
these lines are the inlined version of symbolic product for the tensor size from its dimensions. Not sure about how the symbolic shape looks like to test it